pkg/scalers/metrics_api_scaler.go

```diff
@@ -278,8 +300,156 @@ func getValueFromYAMLResponse(body []byte, valueLocation string) (float64, error
 	}
 }
 
+func (s *metricsAPIScaler) getEndpointsUrlsFromServiceURL(ctx context.Context, serviceURL string) (endpointUrls []string, err error) {
+	// parse service name from s.meta.url
+	url, err := neturl.Parse(serviceURL)
+	if err != nil {
+		s.logger.Error(err, "Failed parsing url for metrics API")
+		return nil, err
+	}
+
+	splittedHost := strings.Split(url.Host, ".")
+	if len(splittedHost) < 2 {
+		return nil, fmt.Errorf("invalid hostname %s : expected at least 2 elements, first being service name and second being the namespace", url.Host)
+	}
+	serviceName := splittedHost[0]
+	namespace := splittedHost[1]
+	podPort := url.Port()
+	// infer port from service scheme when not set explicitly
+	if podPort == "" {
+		if url.Scheme == secureHTTPScheme {
+			podPort = "443"
+		} else {
+			podPort = "80"
+		}
+	}
+	// get service serviceEndpointsSlices
+	serviceEndpointsSlices := &discoveryV1.EndpointSliceList{}
+	serviceNameSelector := labels.NewSelector()
+	serviceNameSelector.Matches(labels.Set(map[string]string{
+		discoveryV1.LabelServiceName: serviceName,
+	}))
+	err = s.kubeClient.List(ctx, serviceEndpointsSlices, &client.ListOptions{
+		LabelSelector: serviceNameSelector,
+		Namespace:     namespace,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var uniqueAddresses []string
+	for _, endpointSlice := range serviceEndpointsSlices.Items {
+		for _, eps := range endpointSlice.Endpoints {
+			// as suggested in https://github.com/kedacore/keda/pull/6565#discussion_r2395073047, make sure we take endpoint into account
+			// only when it's ready
+			if eps.Conditions.Ready == nil || !*eps.Conditions.Ready {
```

Comment:
it is a bit counterintuitive but `Conditions.Ready == nil` should be treated as "ready", see also [api note](https://github.com/kubernetes/api/blob/c397f90a55bd51f3f33e729ae6728c219beab2cf/discovery/v1/types.go#L137-L144)

```suggestion
			if eps.Conditions.Ready != nil && !*eps.Conditions.Ready {
```